[BUG] JitPack build is failing for versions 4.14.0 and 4.15.0

### What is the bug?

The JitPack build for BetterTeams is failing for versions `4.14.0` and `4.15.0`. This means that developers cannot add these newer versions of the plugin as a dependency in their Gradle or Maven projects, as the artifact cannot be downloaded

### Expected Behaviour

The JitPack build should complete successfully for all tagged releases, allowing the artifact to be resolved and used as a dependency in other projects

### How to Reproduce

Create a new Gradle or Maven project
Add BetterTeams `4.15.0` as a dependency
Attempt to build the project. The build will fail because the dependency cannot be found

### Error message

There are two relevant errors.

1. The error when trying to build my project:
```java
> Could not find com.github.booksaw:BetterTeams:4.15.0.
  Searched in the following locations:
    - [https://repo.maven.apache.org/maven2/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom](https://repo.maven.apache.org/maven2/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom)
    - [https://repo.papermc.io/repository/maven-public/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom](https://repo.papermc.io/repository/maven-public/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom)
    - [https://jitpack.io/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom](https://jitpack.io/com/github/booksaw/BetterTeams/4.15.0/BetterTeams-4.15.0.pom)
```
2. The error from the public JitPack build log for version 4.15.0:

```java
[ERROR] COMPILATION ERROR : 
[ERROR] /home/jitpack/build/src/main/java/com/booksaw/betterTeams/score/ScoreManagement.java:[8,32] cannot access com.tcoded.folialib.impl.FoliaImplementation
  bad class file: /home/jitpack/.m2/repository/com/tcoded/FoliaLib/0.5.1/FoliaLib-0.5.1.jar(com/tcoded/folialib/impl/FoliaImplementation.class)
    class file has wrong version 61.0, should be 52.0
```

### Output of /teama debug - **FULL OUTPUT REQUIRED, THE BUG REPORT WILL NOT BE REVIEWED WITHOUT THIS**

not in game bug

### Additional Information

The last version that built successfully on JitPack was 4.13.4. This issue seems to have started with version 4.14.0. Thank you for looking into this

--- a/pom.xml
+++ b/pom.xml
@@ -12,8 +12,8 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <maven.compiler.source>8</maven.compiler.source>
-        <maven.compiler.target>8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
     </properties>
 
     <repositories>
@@ -76,7 +76,7 @@
         <dependency>
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot-api</artifactId>
-            <version>1.16.5-R0.1-SNAPSHOT</version>
+            <version>1.20-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
@@ -157,6 +157,7 @@
             <groupId>net.kyori</groupId>
             <artifactId>adventure-text-minimessage</artifactId>
             <version>4.24.0</version>
+            <scope>compile</scope>
         </dependency>
 
         <!-- https://mvnrepository.com/artifact/net.kyori/adventure-text-serializer-legacy -->
@@ -164,6 +165,7 @@
             <groupId>net.kyori</groupId>
             <artifactId>adventure-text-serializer-legacy</artifactId>
             <version>4.23.0</version>
+            <scope>compile</scope>
         </dependency>
 
         <!-- https://mvnrepository.com/artifact/net.kyori/adventure-text-serializer-bungeecord -->
@@ -171,6 +173,7 @@
             <groupId>net.kyori</groupId>
             <artifactId>adventure-text-serializer-bungeecord</artifactId>
             <version>4.4.1</version>
+            <scope>compile</scope>
         </dependency>
 
         <!-- https://mvnrepository.com/artifact/net.kyori/adventure-platform-bukkit -->
@@ -178,13 +181,7 @@
             <groupId>net.kyori</groupId>
             <artifactId>adventure-platform-bukkit</artifactId>
             <version>4.4.1</version>
-        </dependency>
-
-        <!-- https://mvnrepository.com/artifact/com.google.guava/guava -->
-        <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-            <version>33.4.8-jre</version>
+            <scope>compile</scope>
         </dependency>
 
         <!-- JUnit 5 -->
@@ -251,6 +248,11 @@
                             <!-- !! MAKE SURE TO CHANGE THIS TO YOUR PLUGIN'S GROUP & PLUGIN NAME !! -->
                             <shadedPattern>com.booksaw.betterTeams.lib.folialib</shadedPattern>
                         </relocation>
+                        <!-- Adventure platforms -->
+                        <relocation>
+                            <pattern>net.kyori</pattern>
+                            <shadedPattern>com.booksaw.betterTeams.lib.net.kyori</shadedPattern>
+                        </relocation>
                     </relocations>
                     <filters>
                         <filter>
@@ -276,8 +278,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.14.0</version>
                 <configuration>
-                    <source>8</source>
-                    <target>8</target>
+                    <source>17</source>
+                    <target>17</target>
                     <compilerArgs>
                         <arg>-Xlint:deprecation</arg>
                         <arg>-Xlint:unchecked</arg>
@@ -297,7 +299,7 @@
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.11.2</version>
                 <configuration>
-                    <reportOutputDirectory>docs/javadocs</reportOutputDirectory>
+                    <outputDirectory>docs/javadocs</outputDirectory>
                     <sourcepath>src/main/java</sourcepath>
                 </configuration>
             </plugin>

--- a/src/main/java/com/booksaw/betterTeams/integrations/placeholder/IndividualTeamPlaceholderProvider.java
+++ b/src/main/java/com/booksaw/betterTeams/integrations/placeholder/IndividualTeamPlaceholderProvider.java
@@ -2,9 +2,9 @@ package com.booksaw.betterTeams.integrations.placeholder;
 
 import com.booksaw.betterTeams.Team;
 import com.booksaw.betterTeams.team.TeamManager;
-import org.apache.commons.lang.ArrayUtils;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Objects;
 import java.util.function.Function;
 
 /**
@@ -18,11 +18,17 @@ public interface IndividualTeamPlaceholderProvider {
 	default String getPlaceholderForTeam(@NotNull Team team, @NotNull Function<TeamManager, String[]> sortingFunction) {
 		// Get the sorted teams using the provided sorting function
 		String[] sortedTeams = sortingFunction.apply(Team.getTeamManager());
+		int position = 0;
 
-		// Find the position of the team in the sorted array
-		int position = ArrayUtils.indexOf(sortedTeams, team.getName()) + 1;
+		if (sortedTeams != null) {
+			for (int i = 0; i < sortedTeams.length; i++) {
+				if (Objects.equals(sortedTeams[i], team.getName())) {
+					position = i + 1;
+					break;
+				}
+			}
+		}
 
-		// Return the position as a string
-		return position + "";
+		return String.valueOf(position);
 	}
 }

--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -5,7 +5,7 @@ description: A plugin to create teams with other players
 main: com.booksaw.betterTeams.Main
 website: https://www.spigotmc.org/resources/better-teams.17129/
 folia-supported: true
-api-version: 1.13
+api-version: 1.20
 
 softdepend:
   - PlaceholderAPI

